fontforge/svg.c

```diff
@@ -766,60 +766,75 @@ SplineChar *SCHasSubs(SplineChar *sc, uint32 tag) {
 	if ( pst->type==pst_substitution &&
 		FeatureTagInFeatureScriptList(tag,pst->subtable->lookup->features) )
 return( SFGetChar(sc->parent,-1,pst->u.subs.variant));
+
     }
 return( NULL );
 }
 
+char *GetLigature(SplineChar *sc) {
+    PST *best = HasLigature(sc);
+    if ( best != NULL ) {
+    	int32 univals[50];
+        char *buf = "";
+        int i, c;
+		c = LigCnt(sc->parent,best,univals,sizeof(univals)/sizeof(univals[0]));
+		for ( i=0; i<c; ++i ) {
+		    if ( (univals[i]>='A' && univals[i]<='Z') ||  (univals[i]>='a' && univals[i]<='z') || (univals[i] >= '0' && univals[i] <= '9') || univals[i] == '_' || univals[i] == '-')
+		    	asprintf(&buf,"%s%c", buf, univals[i]);
+		    else
+		    	asprintf(&buf, "%s&#x%x;", buf, (unsigned int) univals[i]);
+		}
+        return (  copy(buf)  );
+    }
+    return ( NULL );
+}
+
+
 static void svg_scdump(FILE *file, SplineChar *sc,int defwid, int encuni, int vs,int layer) {
-    PST *best=NULL;
     const unichar_t *alt;
     int32 univals[50];
     int i, c;
 
-    best = HasLigature(sc);
     if ( sc->comment!=NULL ) {
-	fprintf( file, "\n<!--\n%s\n-->\n",sc->comment );
+		fprintf( file, "\n<!--\n%s\n-->\n",sc->comment );
     }
     fprintf(file,"    <glyph glyph-name=\"%s\" ",sc->name );
-    if ( best!=NULL ) {
-	c = LigCnt(sc->parent,best,univals,sizeof(univals)/sizeof(univals[0]));
-	fputs("unicode=\"",file);
-	for ( i=0; i<c; ++i )
-	    if ( univals[i]>='A' && univals[i]<='z' )
-		putc(univals[i],file);
-	    else
-		fprintf(file,"&#x%x;", (unsigned int) univals[i]);
-	fputs("\" ",file);
+
+    char *ligature = GetLigature(sc);
+
+    if ( ligature!=NULL ) {
+	    fprintf(file,"unicode=\"%s\" ", ligature);
     } else if ( encuni!=-1 && encuni<0x110000 ) {
-	if ( encuni!=0x9 &&
-		encuni!=0xa &&
-		encuni!=0xd &&
-		!(encuni>=0x20 && encuni<=0xd7ff) &&
-		!(encuni>=0xe000 && encuni<=0xfffd) &&
-		!(encuni>=0x10000 && encuni<=0x10ffff) )
-	    /* Not allowed in XML */;
-	else if ( (encuni>=0x7f && encuni<=0x84) ||
-		  (encuni>=0x86 && encuni<=0x9f) ||
-		  (encuni>=0xfdd0 && encuni<=0xfddf) ||
-		  (encuni&0xffff)==0xfffe ||
-		  (encuni&0xffff)==0xffff )
-	    /* Not recommended in XML */;
-	else if ( encuni>=32 && encuni<127 &&
-		encuni!='"' && encuni!='&' &&
-		encuni!='<' && encuni!='>' )
-	    fprintf( file, "unicode=\"%c\" ", encuni);
-	else if ( encuni<0x10000 &&
-		( isarabisolated(encuni) || isarabinitial(encuni) || isarabmedial(encuni) || isarabfinal(encuni) ) &&
-		unicode_alternates[encuni>>8]!=NULL &&
-		(alt = unicode_alternates[encuni>>8][encuni&0xff])!=NULL &&
-		alt[1]=='\0' )
-	    /* For arabic forms use the base representation in the 0600 block */
-	    fprintf( file, "unicode=\"&#x%x;\" ", alt[0]);
-	else if ( vs!=-1 )
-	    fprintf( file, "unicode=\"&#x%x;\" ", vs);
-	else
-	    fprintf( file, "unicode=\"&#x%x;\" ", encuni);
+		if ( encuni!=0x9 &&
```

Comment:
Same reformatting caveat here.

Reply:
There are a mix with tabs and spaces, so my IDE was confused i bit.
Agree, will be fixed